test/hotspot/jtreg/gc/arguments/TestUseCompressedOopsFlagsWithUlimit.java

```diff
@@ -0,0 +1,95 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package gc.arguments;
+
+/*
+ * @test TestUseCompressedOopsFlagsWithUlimit
+ * @bug 8280761
+ * @summary Verify correct UseCompressedOops when MaxRAM and MaxRAMPercentage
+ * are specified with ulimit -v.
+ * @library /test/lib
+ * @library /
+ * @requires vm.bits == "64"
+ * @requires os.family != "aix" & os.family != "windows"
+ * @run driver gc.arguments.TestUseCompressedOopsFlagsWithUlimit
+ */
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+
+public class TestUseCompressedOopsFlagsWithUlimit {
+
+  private static void checkFlag(long maxram, int maxrampercent, boolean forcecoop, boolean expectcoop) throws Exception {
+
+    ArrayList<String> args = new ArrayList<String>();
+    args.add("-XX:MaxRAM=" + maxram);
+    args.add("-XX:MaxRAMPercentage=" + maxrampercent);
+    if (forcecoop) {
+      args.add("-XX:+UseCompressedOops");
+    }
+
+    args.add("-XX:+PrintFlagsFinal");
+    args.add("-version");
+
+    String cmd = ProcessTools.getCommandLine(ProcessTools.createTestJvm(args.toArray(String[]::new)));
+    ProcessBuilder pb = new ProcessBuilder("sh", "-c", "ulimit -v 10485760;" + cmd);
+    OutputAnalyzer output = new OutputAnalyzer(pb.start());
+    output.shouldHaveExitValue(0);
+    String stdout = output.getStdout();
+
+    boolean actualcoop = getFlagBoolValue("UseCompressedOops", stdout);
+    if (actualcoop != expectcoop) {
+      throw new RuntimeException("UseCompressedOops set to " + actualcoop +
+        ", expected " + expectcoop + " when running with the following flags: " + Arrays.asList(args).toString());
+    }
+  }
+
+  private static boolean getFlagBoolValue(String flag, String where) {
+    Matcher m = Pattern.compile(flag + "\\s+:?= (true|false)").matcher(where);
+    if (!m.find()) {
+      throw new RuntimeException("Could not find value for flag " + flag + " in output string");
+    }
+    return m.group(1).equals("true");
+  }
+
+  public static void main(String args[]) throws Exception {
+    // Tests
+    // 1. Verify that UseCompressedOops Ergo follows ulimit -v setting.
+    // 2. Verify that UseCompressedOops forces compressed oops limit even
+    //    when ulimit -v are specified.
```

Comment:
I understand the first bullet but not the second. Looking at the reproducers in the JBS ticket, `UseCompressedOops` should not be provided; it's read only.

Reply:
This looks like an additional (unrelated) test to make sure it is overridden if necessary imho. It seems a bit forced though because there is no test that actually verifies that `UseCompressedOops` is `false` later.
I'm good with removing the test case.

Reply:
> This looks like an additional (unrelated) test to make sure it is overridden if necessary imho.

That is correct.  
Do you mean that the following code should be removed?  

`checkFlag(128 * oneG, 100, true, true);`

Reply:
I understand.  
I removed this test patern.